### ISSUE TITLE
[FR] Retour should respect Craft user site permissions

### ISSUE THREAD

**Issue by steverowling** [nystudio107/craft-retour#318]
### Is your feature request related to a problem? Please describe.
Retour ignores Craft user site permissions. For example, if you have a multisite Craft set up with three sites: A, B and C and site editor groups that only have access to one site each: `siteAEditors`, `siteBEditors` and `siteCEditors`. If you give someone in the `siteAEditors` editors group access to Retour as well, they will be able to see and edit redirects for all three sites, instead of just site A. Everywhere else in Craft, they will only be able to see and edit content for Site A (as determined by section and asset permissions), so it feels wrong, and potentially very dangerous, for someone who should only be able to affect one site, have access to edit and manage redirects for all sites in an installation, regardless of their individual Craft site access permissions.

### Describe the solution you would like
Retour should respect a user's site access permissions and only allow them to view and edit redirects for sites that they have been explicitly granted access to.

### Additional context
The context for this FR is an external SEO consultant who has asked to have access to Retour to set up redirects regularly based on GSC data for one site out of a multi-site set up. They should have no access to the other site in the Craft set up and Craft's site permissions allow us to restrict them in this way. However, Retour does not respect Craft's site access permissions for a user and just shows them the redirects for all sites in the install.

**Comment by khalwat** [comment]
So this is tough, because there may not be a direct correlation between sites you want a given group to be able to access in Craft for content editing purposes, but you might want them to be able to access those site in Retour for redirects.

For example, you may not want your SEO consultant to be able to edit any content in Craft for Site A, but you might want them to be able to create redirects for it.

Any thoughts on this? Or are you good with just a blanket restriction based on the Craft sites permissions?

**Comment by steverowling** [comment]
> So this is tough, because there may not be a direct correlation between sites you want a given group to be able to access in Craft for content editing purposes, but you might want them to be able to access those site in Retour for redirects.
> 
> For example, you may not want your SEO consultant to be able to edit any content in Craft for Site A, but you might want them to be able to create redirects for it.
> 
> Any thoughts on this? Or are you good with just a blanket restriction based on the Craft sites permissions?

Interesting points. From my point of view, the use case I have is for a blanket restriction based on Craft sites permissions.

With that in place, you can fairly easily restrict an external consultant to being able to only access specific sites in Retour. Whether or not you also give them access to be able to edit content in those sites is optional based on the section permissions. If someone is only given access to Site A in the Craft site permissions, and also given access to Retour, but nothing else, then they should be able to manage redirects in Retour for that site, but wouldn't be able to access the content, since they haven't been given access to any of the sections for that site.

So, yeah, I think blanket support for Craft sites permissions should cover most use cases.

**Comment by steverowling** [comment]
I forgot to mention, that the site in question is still on Craft 4, so the FR is for the Craft 4 version of Retour as well as the Craft 5 version.

**Comment by khalwat** [comment]
Right, I'm still backporting everything (features, bugfixes, etc.) to all of Craft 5, Craft 4, and Craft 3.

**Comment by JayBox325** [comment]
I'm just coming to this now as this plugin is exactly what I need, but I have a multi-site setup where a user for Site A shouldn't see anything about Site B or Site C.

<img width="763" alt="Image" src="https://github.com/user-attachments/assets/458313b7-9d11-49fb-bd52-3ed82b20d194" />

I'd just need the "all sites" option removing and ideally if there is only one site available, removing the option to switch between them as this will become a dropdown with one item, similar to this discussion: https://github.com/craftcms/cms/discussions/16526

**Comment by bramnijssen** [comment]
+1 for this feature
